core/domain/stats_jobs_one_off_test.py

```diff
@@ -28,11 +34,663 @@
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=1).put()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id5',
+            exp_id=self.exp_id,
+            exp_version=2,
+            state_name='b',
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=2).put()
+
+        state_stats_dict = {
+            'total_answers_count_v1': 3,
+            'total_answers_count_v2': 9,
+            'useful_feedback_count_v1': 3,
+            'useful_feedback_count_v2': 9,
+            'total_hit_count_v1': 3,
+            'total_hit_count_v2': 0,
+            'first_hit_count_v1': 3,
+            'first_hit_count_v2': 0,
+            'num_times_solution_viewed_v2': 0,
+            'num_completions_v1': 3,
+            'num_completions_v2': 0
+        }
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, self.exp_version, 0, 0, 0, 0, 0, 0,
+            {
+                self.state: state_stats_dict
+            })
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, 2, 0, 0, 0, 0, 0, 0,
+            {
+                self.state: state_stats_dict
+            })
+
+    def test_standard_operation(self):
+        def mock(x, y, z): #pylint: disable=unused-argument,invalid-name
+            return [{'commit_cmds':[]}, {
+                'commit_cmds': [{'cmd': exp_domain.CMD_RENAME_STATE,
+                                 'old_state_name': self.state,
+                                 'new_state_name': 'b'}]}]
+        with self.swap(
+            base_models.VersionedModel, 'get_snapshots_metadata',
+            types.MethodType(mock, base_models.VersionedModel)):
+            job_id = (
+                stats_jobs_one_off.RecomputeStatistics.create_new())
+            stats_jobs_one_off.RecomputeStatistics.enqueue(job_id)
+
+            self.assertEqual(
+                self.count_jobs_in_taskqueue(
+                    taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+            self.process_and_flush_pending_tasks()
+
+        model_id = stats_models.ExplorationStatsModel.get_entity_id(
+            self.exp_id, str(self.exp_version))
+        model = stats_models.ExplorationStatsModel.get(model_id)
+        state_stats = model.state_stats_mapping[self.state]
+        self.assertEqual(state_stats['total_answers_count_v2'], 3)
+        self.assertEqual(state_stats['useful_feedback_count_v2'], 2)
+
+        model_id = stats_models.ExplorationStatsModel.get_entity_id(
+            self.exp_id, '2')
+        model = stats_models.ExplorationStatsModel.get(model_id)
+        state_stats = model.state_stats_mapping['b']
+        self.assertEqual(state_stats['total_answers_count_v2'], 4)
+        self.assertEqual(state_stats['useful_feedback_count_v2'], 3)
+
+
+
```

Comment:
Drop one of these lines -- we separate classes by 2 blank lines, and methods by 1 blank line.

Reply:
Done.